sympy/functions/special/gamma_functions.py

```diff
@@ -319,6 +319,8 @@ def eval(cls, a, x):
 
         # Special values.
         if a.is_Number:
+            if a is S.Zero and x.is_number:
+                return S.Infinity
```

Comment:
This doesn't seem right. What is the `x.is_number` check for?

Reply:
I tried to somewhat justify this in a comment above https://github.com/sympy/sympy/pull/23190#discussion_r819197236.

The special case addition is good and helping a lot of limits as the discussions here show but we don't want this right (which obviously is a result of getting rig of the `is_number` check )?
```
>>> lowergamma(0, x)
oo
```
The only case which I plan to affect is when `a = 0` and x is either `0` or non zero 
`lowergamma(0, 0) = 0` and `lowergamma(0, non-zero) = oo` as shown in the Op (wolfram results)  . What Kalevi pointed out is that most limits which we encounter would be dealing with either `a` or `x` being `0` or both , hence though this wasn't a pr addressing limits originally but it sort of fell into that path , hence his doubts were quite valid !

I hope I conveyed the point across !

Reply:
Read here:
https://docs.sympy.org/latest/guides/assumptions.html

You need to bear in mind that `x.is_zero` can also give None. There are 3 cases to handle: True, False and None. What should happen if `x` is an expression that is not known to be zero or nonzero? Also what if `x` is `Symbol('x', nonzero=True)`?

The check could maybe be `if a is S.Zero and x.is_zero is False`.

Reply:
Oops my bad , cases like this just skipped my mind `Symbol('x', nonzero=True)` . Thanks for pointing this , will keep in mind for future work 

And yeah this would do it !
>if a is S.Zero and x.is_zero is False

Reply:
What should happen if `x` is infinite?
```python
In [57]: lowergamma(0, oo).evalf()
...
ValueError: gamma function pole
```
Again what if `x` is a symbol that is possibly infinite?

Numerically evaluating close to `0, zoo` gives different answers for different phases:
```python
In [61]: for u1 in [1, I, -1, -I]:
    ...:     for u2 in [1, I, -1, -I]:
    ...:         print(lowergamma(u1*1e-10, u2*1e10).evalf(5))
    ...: 
1.0000e+10
1.0e+10 - 0.21257*I
1.0778e+4342944809 + 3.3859e+4342944799*I
1.0e+10 + 0.21257*I
-0.57722 - 1.0e+10*I
-0.63206 - 1.0e+10*I
1.0778e+4342944809 + 2.4816e+4342944800*I
-0.6024 - 1.0e+10*I
-1.0000e+10
-1.0e+10 - 0.18292*I
1.0778e+4342944809 - 3.3859e+4342944799*I
-1.0e+10 + 0.18292*I
-0.57722 + 1.0e+10*I
-0.6024 + 1.0e+10*I
1.0778e+4342944809 - 2.4816e+4342944800*I
-0.63206 + 1.0e+10*I
```
Only one of these sixteen combinations resembles `oo`. We only hit this code path if `a.is_Number` so `a` would have to be real but still this doesn't look like it is approaching `oo` as `a` goes to zero:
```python
In [64]: lowergamma(-1e-10, 1e10)
Out[64]: -10000000000.5772
```

Reply:
Well , as `lowergamma(a, x)` is better defined as `gamma(a) - uppergamma(a, x)`  as mentioned clearly in sympy and wolfram docs too . Hence we have 
```
>>> lowergamma(0, x).rewrite(uppergamma)
-expint(1, x) + zoo
>>> expint(1, oo).n()
0
>>> expint(1, 1).n()
0.219383934395520
>>> expint(1, -1).n()
-1.89511781635594 - 3.14159265358979*I
>>> expint(1, -oo).n()
-oo - 3.14159265358979*I
```
Now any of these values subtracted from `zoo` obviously gives `zoo ` . I went with `oo` just because of what's shown below and because of what wolfram says !
```
>>> lowergamma(0, 1).n() # these give oo and not zoo 
oo 
```
The best part here is wolfram returns `oo` for something like `lowergamma(0, 1)/lowergamma(0, oo) / lowergamma(0, -oo) `and `zoo` for `lowergamma(-1, 1) / lowergamma(-1, oo) / lowergamma(-1, -oo)`

The former ones have been shown in the Op , the later one is here  
![image](https://user-images.githubusercontent.com/87052487/156926372-56122b83-9256-43a7-af12-78161b05cd24.png)

I also checked out what `diofant` return for more clarity on this and it returns `zoo` for something like `lowergamma(0, 1)` .
I'm not too sure about values where `a` is around/tending to origin but when it is at origin , I'm quite sure comparing with other CAS and wiki that it deserves `oo` or `zoo `

Reply:
Also not forgetting to answer your question ,we have the following currently which looks OK to me !
>>> x = Symbol('x', infinite=True)
>>> lowergamma(0, x)
oo

Reply:
That's definitely incorrect because `x` might represent `-oo`. I think probably `zoo` is the right result here.

Reply:
Yeah but we have the following so having x as -oo shouldn't be an issue but I still see `zoo` being a better result here 
![image](https://user-images.githubusercontent.com/87052487/156179700-c1d0aa05-e21c-48c8-b2ad-3529dfa89308.png) .
I hope you also went through the comment above https://github.com/sympy/sympy/pull/23190#discussion_r820238810, which sort of explains why I ended up using `oo` instead of `zoo`